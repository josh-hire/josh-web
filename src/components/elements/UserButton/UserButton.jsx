import PropTypes from 'prop-types';
import { UnstyledButton, Group, Avatar, Text, createStyles } from '@mantine/core';

export const useStyles = createStyles((theme) => ({
  user: {
    display: 'block',
    borderRadius: '8px',
    width: '100%',
    padding: theme.spacing.md,
    color: theme.colorScheme === 'dark' ? theme.colors.dark[0] : theme.black,

    '&:hover': {
      backgroundColor: theme.colorScheme === 'dark' ? theme.colors.dark[8] : theme.colors.gray[0],
    },
  },
}));
export function UserButton({ image, name, email, icon, ...others }) {
  const { classes } = useStyles();

  return (
    <UnstyledButton className={classes.user} {...others}>
      <Group>
        <Avatar radius="xl" src={image} />
        <Group>
          <div style={{ flex: 1 }}>
            <Text size="sm" weight={500}>
              {name}
            </Text>

            <Text color="dimmed" size="xs">
              {email}
            </Text>
          </div>

          {icon}
        </Group>
      </Group>
    </UnstyledButton>
  );
}

UserButton.defaultProps = {
  email: '',
  icon: null,
  image: '',
  name: '',
};

UserButton.propTypes = {
  email: PropTypes.string,
  icon: PropTypes.node,
  image: PropTypes.string,
  name: PropTypes.string,
};
