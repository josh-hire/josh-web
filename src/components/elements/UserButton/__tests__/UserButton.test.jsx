import '@testing-library/jest-dom';
import { UserButton } from '../UserButton';
import ShallowRenderer from 'react-test-renderer/shallow';
import '@testing-library/jest-dom';

describe('src/component/elements/UserButton', ()=>{
  test('UserButton render',()=>{
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<UserButton />);
    expect(tree).toMatchSnapshot();
  });
});
