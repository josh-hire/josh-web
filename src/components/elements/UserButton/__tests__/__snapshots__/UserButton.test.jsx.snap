// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/component/elements/UserButton UserButton render 1`] = `
<@mantine/core/UnstyledButton
  className="mantine-6th54c"
>
  <@mantine/core/Group>
    <@mantine/core/Avatar
      radius="xl"
      src=""
    />
    <@mantine/core/Group>
      <div
        style={
          {
            "flex": 1,
          }
        }
      >
        <@mantine/core/Text
          size="sm"
          weight={500}
        >
          
        </@mantine/core/Text>
        <@mantine/core/Text
          color="dimmed"
          size="xs"
        >
          
        </@mantine/core/Text>
      </div>
    </@mantine/core/Group>
  </@mantine/core/Group>
</@mantine/core/UnstyledButton>
`;
